frank-flow/src/frontend/src/app/shared/services/flow-namespace.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { CurrentFileService } from './current-file.service';
+import { File } from '../../shared/models/file.model';
+import { FlowStructure } from '../models/flow-structure.model';
+import { MonacoEditorComponent } from 'src/app/editor/monaco-editor/monaco-editor.component';
+import { FlowStructureService } from './flow-structure.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FlowNamespaceService {
+  public monacoEditorComponent?: MonacoEditorComponent;
+  private currentFile!: File;
+  private flowStructure!: FlowStructure;
+
+  constructor(
+    private currentFileService: CurrentFileService,
+    private flowStructureService: FlowStructureService
+  ) {
+    this.getCurrentFile();
+  }
+
+  getCurrentFile(): void {
+    this.currentFileService.currentFileObservable.subscribe({
+      next: (currentFile: File): void => {
+        this.currentFile = currentFile;
+        if (currentFile.flowStructure) {
+          this.flowStructure = currentFile.flowStructure;
+        }
+      },
+    });
+  }
```

Comment:
Get current file should get the current file. The name of the function is misleading because it only gets the flowStructure when there is one.

You could just set the currentFile, and use it's flowStructure further one (with a `?` of course) 

If you see it somewhere else in the code, this should also be changed.